src/evaluate/evaluator/token_classification.py

```diff
@@ -0,0 +1,256 @@
+        ):
+            pipe = pipeline(self.task, model=model_or_pipeline, tokenizer=tokenizer, **kwargs)
+        elif isinstance(model_or_pipeline, Pipeline):
+            pipe = pipeline(self.task, model=model_or_pipeline.model, tokenizer=model_or_pipeline.tokenizer, **kwargs)
+        else:
+            if model_or_pipeline is None:
+                pipe = pipeline(self.task, **kwargs)
+            else:
+                pipe = model_or_pipeline
+            if tokenizer is not None:
+                logger.warning("Ignoring the value of the `tokenizer` argument.")
+        if pipe.task != self.task:
+            raise ValueError(
+                f"Incompatible `model_or_pipeline`. Please specify `model_or_pipeline` compatible with the `{self.task}` task."
+            )
+
+        # Prepare metric.
+        if metric is None:
+            if self.default_metric_name is None:
+                raise ValueError(
+                    "`Evaluator` doesn't specify a default metric. Please specify a valid `metric` argument."
+                )
+            metric = load(self.default_metric_name)
+        elif isinstance(metric, str):
+            metric = load(metric)
+
+        # Prepare reference.
+        features = data.features
+
+        # If the labels are of type ClassLabel, they are already integers and we have the map stored somewhere.
+        # Otherwise, we have to get the list of labels manually.
+        labels_are_int = isinstance(features[label_column].feature, ClassLabel)
+        if labels_are_int:
+            label_list = features[label_column].feature.names  # list of string labels
+            ref_to_labels = {i: label for i, label in enumerate(label_list)}
+        elif features[label_column].feature.dtype.startswith("int"):
+            raise NotImplementedError(
+                "References provided as int, but the reference column is not instanciated as a Sequence of ClassLabel."
+            )
+        else:
+            # In the event the labels are not a `Sequence[ClassLabel]`, we will need to go through the dataset to get the
+            # unique labels.
+            # An example is labels as ["PER", "PER", "O", "LOC", "O", "LOC", "O"], e.g. in polyglot_ner dataset
+            # `ref_to_labels` will map labels to labels, here {"LOC": "LOC", "PER": "PER", "O": "O"}
+            # Normally `ref_to_labels` would just be e.g. {0: "O", 1: "LOC", 2: "PER"}
+            unique_labels = set()
+            for label in data[label_column]:
+                unique_labels = unique_labels | set(label)
+            ref_to_labels = {label: label for label in unique_labels}
+
+        # Core computations.
+        references = [[ref_to_labels[l] for l in label] for label in data[label_column]]
+        predictions = self._compute_predictions(pipe, data[input_column], label_mapping=label_mapping, join_by=join_by)
+        result = metric.compute(predictions=predictions, references=references)
```

Comment:
This is the only unique part of the `compute` function, it would make sense to only define `compute_predictions_and_references` in child classes and have shared pre- and post-processing defined in the parent class to avoid duplication.